internal/domain/users/store_test.go

```
@@ -764,6 +765,220 @@ var _ = Describe("UsersStore", func() {
 			})
 		})
 	})
+
+	Describe("Assigning, Getting and Removing User Roles", func() {
+		var (
+			userRolesStore       userroles.Store
+			testUserRole         userroles.UserRole
+			testUserAssignedRole users.UserToRole
+		)
+
+		BeforeEach(func() {
+			setUpUserRolesDependentData(db)
+			userRolesStore = userroles.CreateUserRolesStore(db)
+			testUserRole = userroles.UserRole{
+				ID:   int32(20),
+				Name: "test_user_role_name",
+				CreatedByUserID: sql.NullInt32{
+					Int32: testUserID,
+					Valid: true,
+				},
+				AccountID:     testAccountID,
+				Description:   sql.NullString{String: "test_desc", Valid: true},
+				PermissionIDs: []int32{1, 20},
+			}
+
+			testUserAssignedRole = users.UserToRole{
+				RoleID: testUserRole.ID,
+				UserID: testUserID,
+			}
+		})
+
+		AfterEach(func() {
+			err = utils.TruncateTables(db)
+		})
+
+		Describe("AddRolesForUser", func() {
```

Please add a use case where multiple valid userroles are added to a single user.

---

What would happen when a user is already assigned with a userRole, and we are trying to add the same user role again in the list?

In this case, I would expect us ignore the error, and keep single record in the db to ensure the corresponding user_id & role_id is a valid record in db.

---

Please add test, and adjust the implementation on the sql query we are issuing.

---

Done, please verify